src/main/java/org/eclipse/emfcloud/jackson/annotations/JsonAnnotations.java

```diff
@@ -70,45 +70,41 @@ public static boolean shouldIgnore(final EStructuralFeature feature) {
    }
 
    /**
-    * Returns the property that should be use to store the type information of the classifier.
+    * Returns the property that should be use to store the type information of the eClass.
     *
-    * @param classifier any classifier
+    * @param eClass any eClass
     * @return the type information property
     */
    @SuppressWarnings("checkstyle:cyclomaticComplexity")
-   public static EcoreTypeInfo getTypeProperty(final EClassifier classifier) {
-      String property = getValue(classifier, "JsonType", "property");
-      String use = getValue(classifier, "JsonType", "use");
-
-      ValueReader<String, EClass> valueReader = EcoreTypeInfo.DEFAULT_VALUE_READER;
-      ValueWriter<EClass, String> valueWriter = EcoreTypeInfo.DEFAULT_VALUE_WRITER;
+   public static EcoreTypeInfo getTypeProperty(final EClass eClass,
```

Comment:
I see the reason for the adaptation. But as this is a public api method I would suggest to keep the old signature in order to delegate to the new method with the previous parameters. Also a check that the `EClassifier` is an `EClass` would make sense before delegating it. What do you think?

Reply:
Alright, I will revert this change.

Reply:
@janhicken Thank you! I think I may have been a bit unclear but could you do the null check on the reader/writer argument and default to the DEFAULT_VALUE_READER and DEFAULT_VALUE_WRITER in case null arguments are given?

Reply:
Wouldn't that duplicate the behaviour `EcoreTypeInfo`'s [constructor](https://github.com/eclipse-emfcloud/emfjson-jackson/blob/master/src/main/java/org/eclipse/emfcloud/jackson/annotations/EcoreTypeInfo.java#L68)?

Reply:
Ah, you are absolutely right, seems like I overlooked that somehow the second time around. Thank you!